tests/DI/NeonAdapter.phpt

```diff
@@ -134,3 +134,76 @@ Assert::equal([
 		[5]
 	),
 ], $data);
+
+
+$data = $config->load('files/neonAdapter.entity.neon');
+$config->save($data, TEMP_FILE);
+Assert::match(<<<EOD
+# generated by Nette
+
+- ent(1)
+- ent(2)inner(3, 4)
+- ent(3)inner(5)
+EOD
+, preg_replace('#(\t*\n){3,}#', "\n\n", file_get_contents(TEMP_FILE)));
+
+
+$data = $config->load('files/neonAdapter.save.neon');
+$config->save($data, TEMP_FILE);
+Assert::match(<<<'EOD'
+# generated by Nette
+
+parameters:
+	class: Ipsum
+
+services:
+	referencedService: @one
+	referencedServiceWithSetup:
+		factory: @one
+		setup:
+			- $x(10)
+
+	serviceAsParam: Ipsum(@one)
+	calledService: @one()
+	calledServiceWithArgs: @one(1)
+	calledServiceAsParam: Ipsum(@one())
+	calledServiceWithArgsAsParam: Ipsum(@one(1))
+	one:
+		class: %class%
+		arguments:
+			- 1
+
+	two:
+		class: %class%(1)
+
+	three:
+		class: Lorem
+		create: Factory::createLorem
+		arguments:
+			- 1
+
+	four:
+		create: Factory::createLorem(1)
+
+	five:
+		create: [Factory, createLorem](1)
+
+	six: Factory::createLorem(1)
+	seven: @factory
+	eight: @factory()
+	nine:
+		- @three
+		- foo
+
+	stdClass: stdClass
+	factory: Lorem
+	rich1: Lorem(1)foo()
+	rich2:
+		create: Lorem(Ipsum(@one))foo(1)
+
+	rich3: Factory::createLorem(1)foo()
```

Comment:
Is not better `Factory::createLorem(1)::foo()` than `Factory::createLorem(1)foo()`?